documentation-versioning.rst

```diff
@@ -0,0 +1,71 @@
+.. post:: July 23, 2019
+   :tags: vcs, versioning
+   :author: Manuel
+   :location: BCN
+
+.. meta::
+   :description lang=en:
+
+      Recommendations about how to do documentation versioning.
+
+Documentation Versioning: Good Practices
+========================================
+
+Project releases and documentation versions are strongly related each other.
+We usually want to have a release ``1.0.3`` with it's own documentation version ``1.0.3`` that's specific for that release.
+
+In practice, when we make a new release of our software,
+we create a new tag in our :abbr:`VCS` for this release and activate this new version on Read the Docs.
+At this point, Read the Docs will build and publish the new documentation version ``1.0.3`` for us.
+
+This approach of "VCS tag" mapped to a "Read the Docs version" works in most of cases.
+Although, there could be some problems that may arise in time and that we can prevent using a different strategy.
+
+
+Problems of using tags for documentation versions
+-------------------------------------------------
+
+The main problem of using VCS tags to build documentation versions is that they are static.
+This means, that once the tag is created it will never be modified.
+
+This may not seems like a problem at first sight since that's the main concept of :abbr:`VCS` tags.
+Although, when talking about documentation versions,
+sometimes happen that after making a release there is a typo in the docs, or a small code example that does not work,
+or a sentence that is confusing, etc. and we want to fix it to avoid this inconveniences.
+
+Making a new tag and release just for these small changes does not worth the effort but also it may confuse our users.
+In other words, having a ``1.0.4`` release that only fixes a sentence in the documentation is not optimal.
+
+Another problem that we have found with this approach is when new features are released on Read the Docs.
+Applying this feature to all of our documentation versions --new ones and old ones, it's a problem.
+Similarly, if we want to install a new Sphinx extension in all our versions,
+update one of its settings, change the theme,
+or even change a build config in the `.readthedocs.yml`_,
+we will be "blocked" by the tags being static.
+
+
+Recommended versioning strategy
+-------------------------------
+
+The workflow that we recommend to versioning your documentation is by having release branches instead.
+With this approach, if a reader detects a typo in the documentation,
+we can just fix the typo and commit and push to this ``1.0.x`` release branch.
+By doing this, the documentation will be automatically updated and published on Read the Docs.
+
+Following our example, this strategy would be:
+
+* ``1.0.x`` is a release branch for the ``1.0`` version
+* ``1.0.3`` is a similar tag for your ``1.0.x`` branch, with the latest release
+* ``1.0.0`` is the first tag for your ``1.0.x`` branch.
+
```

Comment:
I feel like we need to expand more on this. This mostly matches what I had here: https://www.ericholscher.com/blog/2016/jul/1/sphinx-and-rtd-for-writers/#recommended-versioning-system -- but what do we do with the tags and old links to `1.0.3` when `1.0.4` is released? I think we need a bit more guidance there.

Reply:
Yes. I took the inspiration from there and my goal was to expend it and make it better. I failed :slightly_smiling_face:

Reply:
I think the key is that if there will be more 1.0.x releases, there should be a release branch and documentation builds off that branch. If you have a typo in the docs or an example in the docs that doesn't work, you just make a new version and the `1.0.x` docs get updated.

It also depends on whether there will be multiple concurrent versions. For example, Django is currently supporting 1.11.x, 2.1.x and 2.2.x and there are large differences between these. Users definitely should be looking at the right docs for their version.

If instead, you just want to have a few old versions of the docs around for users using the old stuff then I think tags are mostly fine. However, again if you're following semver you probably don't need docs for `1.0.3` and `1.0.4`. I think that's worth mentioning in this post!

Reply:
> if you're following semver you probably don't need docs for 1.0.3 and 1.0.4

I mean you have the docs in VCS but you don't need published docs for both when they're basically exactly the same.